crates/core/src/operations/cdc.rs

```diff
@@ -113,7 +113,8 @@ mod tests {
             .await
             .expect("Failed to make a table");
         table.load().await.expect("Failed to reload table");
-        let result = should_write_cdc(table.snapshot().unwrap()).expect("Failed to use table");
+        let result =
+            should_write_cdc(&table.snapshot().unwrap().snapshot).expect("Failed to use table");
```

Comment:
When we were hacking yesterday this kind of accessing into the `snapshot` field of the `EagerSnapshot` is something we should probably be removing.

Is it possible to put a `#[cfg(test)]` around that to keep its isolation into tests?